include/boost/type_traits/detail/is_function_cxx_11.hpp

```diff
@@ -96,7 +96,7 @@ namespace boost {
    template <class Ret, class ...Args BOOST_TT_NOEXCEPT_PARAM>
    struct is_function<Ret(Args..., ...)const volatile && BOOST_TT_NOEXCEPT_DECL> : public true_type {};
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && !defined(_M_ARM) && !defined(_M_ARM64)
```

Comment:
In a purely hypothetical universe where MSVC supports _even more_ architectures, wouldn't that re-surface? Or is that too hypothetical?

Reply:
For now yes ;)  Besides the hypothetical new architecture may need the extra specialisations just as likely as not need them?

More likely IMO is that __vectorcall stops being the same as the default and becomes something different again on arm64 as it is on x64.  Or not.